File: Programmers/Greedy/체육복.py
def solution(n, lost, reserve):
    answer = 0
    cnt = 0
    len_lost = len(lost)
    temp = lost.copy()
    for i in temp:
        print('step', i)
        if i in reserve:
            print('step2', i)
            cnt+=1
            reserve.remove(i)
            lost.remove(i)
            
            
            print('step3', lost, reserve)
            
    print(cnt, lost, reserve)
    
    for i in lost:
        for j in reserve:
            if j==i+1 or j==i-1:
                cnt+=1
                reserve.remove(j)
                break
                 
    print(cnt)
    answer = n - len_lost + cnt
    return answer

File: Programmers/Greedy/체육복.cpp
#include <vector>

using namespace std;

int solution(int n, vector<int> lost, vector<int> reserve) {
        int answer = 0;
    vector<int> uniform(n+2, 1);
    
    for(auto& i : reserve){
        uniform[i]++;
    }
    for(auto& i : lost){
        uniform[i]--;
    }
    for(int i = 1; i < uniform.size(); i++){
        if(uniform[i-1] == 0 && uniform[i] > 1){
            uniform[i-1] = 1;
            uniform[i]--;
        }
        if(uniform[i+1] == 0 && uniform[i]>1){
            uniform[i+1] = 1;
            uniform[i]--;           
        }
    }    
    for(int i = 1; i <= n; i++){
        if(uniform[i]>0) answer++;
    }
    return answer;
}
